File: chroma-league/src/test/java/com/bonepl/chromaleague/rest/http/handlers/GameStatsResponseHandlerTest.java
package com.bonepl.chromaleague.rest.http.handlers;

import com.bonepl.chromaleague.rest.gamestats.GameStats;
import com.bonepl.chromaleague.rest.http.LeagueHttpClientMock;
import org.apache.http.HttpResponse;
import org.junit.jupiter.api.Test;

import java.io.IOException;
import java.net.URISyntaxException;
import java.util.Optional;

import static org.junit.jupiter.api.Assertions.*;

public class GameStatsResponseHandlerTest {

    @Test
    void testGameStatsParsing() throws IOException, URISyntaxException {
        //given
        HttpResponse testResponseFromJSON = LeagueHttpClientMock.createTestResponseFromJSON("json/gamestats.json");

        //when
        Optional<GameStats> gameStats = new GameStatsResponseHandler().handleResponse(testResponseFromJSON);

        //then
        assertTrue(gameStats.isPresent());
        verifyGameStats(gameStats.get());
    }

    public static void verifyGameStats(GameStats gameStats) {
        assertNotNull(gameStats);
        assertEquals("CLASSIC", gameStats.gameMode());
        assertEquals(15.123456789, gameStats.gameTime());
        assertEquals("Infernal", gameStats.mapTerrain());
    }
}

File: test/rest/http/handlers/GameStatsResponseHandler.test.ts
import {describe, expect, test} from 'vitest';
import type {GameStats} from '../../../../src/rest/gamestats/GameStats.js';

describe('GameStatsResponseHandler', () => {
    test('testGameStatsParsing', () => {
        //given
        const json = {
            gameMode: 'CLASSIC',
            gameTime: 15.123456789,
            mapName: 'Map11',
            mapNumber: 11,
            mapTerrain: 'Infernal',
        };

        //when
        const gameStats = json as GameStats;

        //then
        expect(gameStats).toBeDefined();
        verifyGameStats(gameStats);
    });
});

export function verifyGameStats(gameStats: GameStats): void {
    expect(gameStats).toBeDefined();
    expect(gameStats.gameMode).toBe('CLASSIC');
    expect(gameStats.gameTime).toBe(15.123456789);
    expect(gameStats.mapTerrain).toBe('Infernal');
}
